ArduPlane/mode_qrtl.cpp

```diff
@@ -5,27 +5,58 @@
 
 bool ModeQRTL::_enter()
 {
-    // use do_RTL() to setup next_WP_loc
-    plane.do_RTL(plane.home.alt + quadplane.qrtl_alt*100UL);
+    submode = SubMode::RTL;
+    const int32_t RTL_alt_abs = plane.home.alt + quadplane.qrtl_alt*100UL;
     plane.prev_WP_loc = plane.current_loc;
+
     pos_control->set_accel_desired_xy_cmss(Vector2f());
     pos_control->init_xy_controller();
+
+    bool jump_to_pos1 = false;
+    if (quadplane.motors->get_desired_spool_state() == AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED) {
+        // VTOL motors are active, either in VTOL flight or assisted flight
+        Location destination = plane.rally.calc_best_rally_or_home_location(plane.current_loc, RTL_alt_abs);
+
+        const float dist = plane.current_loc.get_distance(destination);
+        const float radius = MAX(fabsf(plane.aparm.loiter_radius), fabsf(plane.g.rtl_radius));
+
+        const float dist_to_climb = quadplane.qrtl_alt - plane.relative_ground_altitude(plane.g.rangefinder_landing);
+        if (dist < 1.5*radius) {
+            // we're close to destination and already running VTOL motors, don't transition and don't climb
+            gcs().send_text(MAV_SEVERITY_INFO,"VTOL position1 d=%.1f r=%.1f", dist, radius);
+            jump_to_pos1 = true;
+
+        } else if (is_positive(dist_to_climb)) {
+            // climb before returning
+            submode = SubMode::climb;
```

Comment:
i don't see where submode RTL is ever set. Probably easiest to default to RTL submode before this section of code

Reply:
First line of the function ;)